pennylane/operation.py

```diff
@@ -538,11 +539,30 @@ def _heisenberg_rep(p):
 
     @classproperty
     def supports_analytic(self):
-        """Returns True if the CV Operation has a defined :meth:`~.CV._heisenberg_rep`
-        static method, indicating that analytic differentiation is supported.
+        """Returns True if the CV Operation has ``grad_method='A'`` and
+        a defined :meth:`~.CV._heisenberg_rep` static method, indicating
+        that analytic differentiation is supported.
         """
-        p = list(range(self.num_params))
-        return self._heisenberg_rep(p) is not None
+        if self.grad_method != 'A':
+            return False
+
+        return self.supports_heisenberg
+
+    @classproperty
+    def supports_heisenberg(self):
+        """Returns True if the CV Operation has
+        a defined :meth:`~.CV._heisenberg_rep` static method, indicating
+        that analytic differentiation is supported if this operation
+        succeeds a gate to be differentiated analytically.
+        """
+        n = self.num_params
+        if self.par_domain == 'A':
+            pars = [np.eye(2)] * n
+        elif self.par_domain == 'N':
+            pars = [0] * n
+        else:
+            pars = [0.0] * n
+        return self._heisenberg_rep(pars) is not None
```

Comment:
Having to construct parameters to be able to call self._heisenberg_rep() and then actually construcitng the rep is not a very elegant way of checking whether a heisenberg rep is in principle supported. Can one not do something like `Operation._heisenberg_rep == type(self)._heisenberg_rep` to test whether the method was overwritten?

This litte example suggests this should be possible:
```
class A:
    def m():
        return 'a'

    def check_if_m_overwritten(self):
        return A.m == type(self).m

class B(A):
    pass

class C(A):
    def m():
        return 'b'


b = B()
c = C()

print(b.check_if_m_overwritten())
print(c.check_if_m_overwritten())
```
Output:
```
True
False
```

Reply:
I agree that this method is not very transparent/elegant. This is an original PL v0.1 method, only showing up as a change due to moving its location in the file. Since this is a separate issue, let's create a new github issue for it and a separate PR.